### PR TITLE
Note that `Rule.use` and `UseEntry` can be functions

### PR DESCRIPTION
**Pull request by glenjamin**
I discovered this earlier this week buried on a github comment somewhere, so wanted to add this into the docs.

I'm not massively happy with the wording here, there weren't any notes about how best to document function arguments or return values in the writing guide, so I figure I'd post this up as-is for feedback.

### CODE REVIEW

**Comment by glenjamin** [comment]
Oh whoops, looks like this overlaps with #2691 and #2694 - although those don't mention that individual `UseEntry` items can also be functions.

**Comment by EugeneHlushko** [comment]
Hello @glenjamin 

I would suggest to continue with your PR after the first one gets merged since it was submitted earlier.

**Comment by EugeneHlushko** [comment]
@glenjamin we had merged #2694 please proceed here after updating the branch

File: src/content/configuration/module.md [hunk 1]
```diff
@@ -404,12 +404,21 @@ module.exports = {
...
 ## `UseEntry`
 
-`object`
+`object | function`
 
 It must have a `loader` property being a string. It is resolved relative to the configuration [`context`](/configuration/entry-context#context) with the loader resolving options ([resolveLoader](/configuration/resolve#resolveloader)).
 
 It can have an `options` property being a string or object. This value is passed to the loader, which should interpret it as loader options.
 
+A `UseEntry` can also be a function which receives the object argument describing the module being loaded, and must return an options object. This can be used to vary the loader options on a per-module basis.
```
Comment (montogeek) [inline src/content/configuration/module.md]: Could you please add an example?

**Comment by glenjamin** [comment]
I've added some subheadings to separate the types a little bit, and put in an example for each.

File: src/content/configuration/module.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ These options determine how the [different types of modules](/concepts/modules)
 
 ## `module.noParse`
 
-`RegExp | [RegExp] | function | string | [string]`
+`RegExp` `[RegExp]`  `function(resource)`  `string` `[string]`
```
Comment (EugeneHlushko) [inline src/content/configuration/module.md]:
```suggestion
`RegExp` `[RegExp]` `function(resource)` `string` `[string]`
```

**Comment by glenjamin** [comment]
Hrm, I can't seem to see a good way to call out the separate docs for each type without having the markdown linter yell at me.

**Comment by EugeneHlushko** [comment]
It actually says that header is duplicated, i suggest you dont use headers there but just go with bold text:
```
```

**Comment by EugeneHlushko** [comment]
@montogeek please have a second look at it mate i thinks its good to go from looking at it.

**Comment by EugeneHlushko** [comment]
@glenjamin please try not to rebase and force-push as commit history is very important for doing quick reviews

**Comment by glenjamin** [comment]
Oh sorry, I thought github was able to keep track!

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**